pcsx2/gui/Panels/GameDatabasePanel.cpp

```diff
@@ -430,15 +434,21 @@ bool Panels::GameDatabasePanel::WriteFieldsToDB() {
 	if (serialBox->GetValue().IsEmpty()) return false;
 
 	Game_Data game;
+	wxString com = compatBox->GetValue();
 	GameDB->findGame(game, serialBox->GetValue());
 
 	game.id = serialBox->GetValue();
 
-	game.writeString(L"Serial",		serialBox->GetValue());
-	game.writeString(L"Name",		nameBox->GetValue());
-	game.writeString(L"Region",		regionBox->GetValue());
-	game.writeString(L"Compat",		compatBox->GetValue());
-	game.writeString(L"[comments]",	commentBox->GetValue());
+	game.writeString(L"Serial", serialBox->GetValue());
+	game.writeString(L"Name", nameBox->GetValue());
+	game.writeString(L"Region", regionBox->GetValue());
+	if (com == "Unknown")          game.writeString(L"Compat", "0");
```

Comment:
Yes, thats why I wrote:
> Or just leave it like that if this is the easiest to write, read and maintain...

How to use a string in a switch even if a string is not well-defined? So without having proper knowledge about the c++ language, compiler and whatever - so being an absolute noob -  I would just "cast" the argument to something I can switch. If some of these strings would have an GetHash() function and implicitly "cast" to this function if used as switch argument including the obligatory collision check I would have a well defined switch (maybe it's slow...). I mean it's not like that the switch statement directly gets translated into asm. Under the hood it can unfold it into an if/else or a lockup. As programmer I shouldn't care what the compiler does when the code is not speed critical. Maybe the switch gets optimized away in any case.

What would be really horrible is if this code gets copy pasted to different locations and somebody afterwards changes the perfect to excellent or adds another case at one location. The compiler could never know that you just missed one spot if you use if/else. In a switch it would know that you forgot one case. An enum would lead to consistent naming.

I mean it's not like I am the inventor of switches. I actually don't like them as they almost always just show momentary laziness and improper OOP programming that always ends with rewriting everything because you need object methods instead of cases. But who wants to code here a compatibility interface and playable/ingame/... classes for one occurrence? So I get why we don't use OOP to solve the problem but considering if/else chains as bad is not unreasonable.

Btw. Java implemented string switches afterwards. In c# it's standard. C++ moves more and more into a modern language due to a lot of updates. Requesting an ISwitchable interface or something like that which gets implemented by most used/modern strings seems to me - noob - possible.